File: 1306-jump-game-iii/1306-jump-game-iii.cpp
class Solution {
public:
    bool canReach(vector<int>& A, int cur) {
        if (cur < 0 || cur >= size(A) || A[cur] < 0)
            return false;
        A[cur] *= -1;
        return !A[cur] || canReach(A, cur + A[cur]) ||
               canReach(A, cur - A[cur]);
    }
};

File: 1306-jump-game-iii/1306-jump-game-iii.py
class Solution:
    def canReach(self, A, cur):
        if cur < 0 or cur >= len(A) or A[cur] < 0:
            return False
        A[cur] *= -1
        return (
            A[cur] == 0
            or self.canReach(A, cur + A[cur])
            or self.canReach(A, cur - A[cur])
        )
